# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [8]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [9]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.3259158 0.3503637 0.1008993 0.6235328 0.8389101 0.6225789 0.2256182
2 0.7786231 0.3741931 0.4289468 0.1530597 0.1603087 0.7589287 0.2681988
  V8        V9         V10       V11       V12       V13       V14      
1 0.4674211 0.87557853 0.2520843 0.8846035 0.6724925 0.8786334 0.3367066
2 0.9840264 0.05101426 0.1804677 0.6145659 0.1728417 0.4210299 0.8920385
  V15       V16       V17       V18         V19        V20      
1 0.6639441 0.9799514 0.6041533 0.918786465 0.60731776 0.8952107
2 0.4410443 0.4786470 0.4174268 0.007422443 0.03242455 0.4126696

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [10]:
df2 <- df1 %>%
  gather()

head(df2)

key value    
1 V1  0.3259158
2 V1  0.7786231
3 V2  0.3503637
4 V2  0.3741931
5 V3  0.1008993
6 V3  0.4289468

## Assignment 2

We will generate some data:

In [11]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [12]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [13]:
df5 <- full_join(df3, df4, by='id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook